### ISSUE TITLE
screen readers don't read labels for radio buttons in bc forms

### ISSUE THREAD

**Issue by johrstrom** [OSC/ondemand#4289]
screen readers don't read labels for radio buttons in bc forms. I think this is because the label is `for` a different ID.

Consider this YAML. the options for 'whole' and 'shared' are read because they have their own label, but the base label 'gpu' is missed entirely.

```yaml
form:
  - gpu
attributes:
  gpu:
    widget: radio_button
    options:
      - ['shared', 'shared']
      - ['whole', 'whole']
    value: 'shared'
    required: true
    help: |
      Request a shared GPU or use the whole GPU.
```

**Comment by Bubballoo3** [comment]
This hopefully will be fixed in https://github.com/bootstrap-ruby/bootstrap_form/pull/779, and so this issue should also include bumping the version to one with that PR included.

**Comment by lcreid** [comment]
Hi folks. Thanks for your patience. I've merged https://github.com/bootstrap-ruby/bootstrap_form/pull/779, but I haven't done a release of the gem. I have a couple of other PRs I'd like to merge before I release a new version. I'd sure appreciate it if you can try using the gem from the GitHub repo to see if it does what you need. I'm not expert on accessibility, so I really appreciate any feedback you can give.

**Comment by lcreid** [comment]
I should have noted that you have to set a configuration option to generate the new markup. Make an initializer file `config/initializers/bootstrap_form.rb` with the following contents, and restart the server:

```
# config/initializers/bootstrap_form.rb

BootstrapForm.configure do |config|
  config.group_around_collections = true
end
```

**Comment by johrstrom** [comment]
> Hi folks. Thanks for your patience.

Thank YOU for your work!  We'll try to carve out some time to see and let you know.

**Comment by Bubballoo3** [comment]
Thanks so much @lcreid! We are by no means accessibility experts either, but as you might have gathered from this issue our main target was just for screenreaders to fully announce the radio buttons without skipping any of the options. 

Unfortunately we are sticking with Rails 7.1 for our upcoming release (see #4718) , so I can't test it out in this repo just yet, but this issue is for our subsequent release, 4.2, which will focus on accessibility. That release is planned for April/May next year, so there is no rush on releasing the updated gem.

In summary, if you have tested it out with a screenreader and it works well then that is everything we know that we need now, and might be able to give more detailed feedback when we begin to dig into the rest of the accessibility items during 4.2 and are able to try out the new version.

**Comment by johrstrom** [comment]
Just for reference - I think I was expecting this to be read through tab navigation as select and text input labels are read through tab navigation.
